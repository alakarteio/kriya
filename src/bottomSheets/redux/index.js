import bottomSheets from './bottomSheets'
import * as actions from './bottomSheets.actions'
import * as selectors from './bottomSheets.selectors'

Object.assign(bottomSheets, actions, selectors)

export * from './bottomSheets.actions'
export * from './bottomSheets.selectors'
export default bottomSheets
