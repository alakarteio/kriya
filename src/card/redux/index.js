export { default } from './card'
export * from './card.actions'
export * from './card.selectors'
