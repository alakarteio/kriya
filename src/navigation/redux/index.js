export { default } from './navigation'
export * from './navigation.actions'
export * from './navigation.selectors'
