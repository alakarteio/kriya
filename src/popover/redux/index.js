export { default } from './popover'
export * from './popover.actions'
export * from './popover.selectors'
