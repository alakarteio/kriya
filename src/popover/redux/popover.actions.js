export const OPEN_POPOVER = 'OPEN_POPOVER'
export const openPopover = (code, contents) =>
  ({ type: OPEN_POPOVER, payload: { code, contents } })

export const CLOSE_POPOVER = 'CLOSE_POPOVER'
export const closePopover = code => ({ type: CLOSE_POPOVER, payload: code })
