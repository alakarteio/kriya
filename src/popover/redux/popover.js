import omit from 'lodash/omit'
import { OPEN_POPOVER, CLOSE_POPOVER } from './popover.actions'

export const initState = { }
export const initAction = { type: 'UNKNOWN' }

export default (state = initState, { type, payload } = initAction) => {
  switch (type) {
    case OPEN_POPOVER:
      return { ...state, [payload.code]: { print: true, contents: payload.contents } }
    case CLOSE_POPOVER:
      return { ...state, [payload]: { ...state[payload], print: false } }
    default: return state
  }
}
