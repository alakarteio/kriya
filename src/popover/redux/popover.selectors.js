export const getPopover = state => state.ui.popover
