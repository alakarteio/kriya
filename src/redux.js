export { default as cards } from './card/redux'
export { default as navigation } from './navigation/redux'
export { default as popover } from './popover/redux'
export { default as bottomSheets } from './bottomSheets/redux'
export { default as tabs } from './tabs/redux'
