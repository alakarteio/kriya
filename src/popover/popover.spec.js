/* eslint-disable
  import/no-extraneous-dependencies,
  react/jsx-filename-extension,
  react/no-children-prop,
  react/prop-types,
*/
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import snap from 'snap'
import { closePopover, CLOSE_POPOVER } from './redux'
import Popover from './popover'
import PopoverContainer from './popover.container'

const Child = ({ print, onClose, contents, ...props }) => <div {...props}>{`custom_component_child${JSON.stringify({ print, onClose, contents })}`}</div>
const Child2 = ({ print, onClose, contents, ...props }) => <div {...props}>{`custom_2${JSON.stringify({ print, onClose, contents })}`}</div>

const snapshot = props => snap(Popover)({ children: <Child />, ...props })

describe('common/Popover', () => {
  describe('graphical (JSX)', () => {
    it('should add custom className', snapshot({ className: 'custom' }))
    it('should add custom style', snapshot({ style: { backgroundColor: 'red' } }))
    it('should add children', snapshot({ children: <Child2 /> }))
    it('should have a default behaviour', snapshot({}))
    it('should be print when asked for and passed it to child', snapshot({ print: true }))
    it('should pass onClose function to child', snapshot({ onClose: () => {} }))
    it('should pass contents to child', snapshot({ contents: { description: 'description' } }))
    it('should call onClose when clicked', () => {
      const onClick = jest.fn()
      const wrapper = mount(
        <Popover children={<Child />} onClose={onClick} />,
      )

      wrapper.find('div').first().simulate('click')
      expect(onClick.mock.calls.length).toBe(1)
    })
  })

  describe('container', () => {
    const snapContainer = (state, props) => {
      const store = createStore(() => ({ ui: { popover: state } }))
      const component = renderer.create(
        <Provider store={store}>
          <PopoverContainer {...props}>
            <Child />
          </PopoverContainer>
        </Provider>,
      )

      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    }

    it('should pass print=false when the popover is not listed in redux', () => snapContainer({ pop1: true }, { code: 'pop2' }))
    it('should pass print=true when the popover IS listed in redux', () => snapContainer({ pop1: true }, { code: 'pop1' }))
    it('should pass onClose callback when closable', () => snapContainer({}, { code: 'pop1', closable: true }))
    it('should pass contents', () => snapContainer({}, { code: 'pop1', contents: { descriptions: 'descriptions' } }))
    it(`should call ${CLOSE_POPOVER} when clicked`, () => {
      const dispatch = jest.fn()
      const store = createStore(() => ({ ui: { popover: { code: 'pop1' } } }))
      store.dispatch = dispatch

      const EmptyChild = () => <div />
      const wrapper = mount(
        <Provider store={store}>
          <PopoverContainer code="pop1" closable>
            <EmptyChild />
          </PopoverContainer>
        </Provider>,
      )

      wrapper.find('div').first().simulate('click')

      expect(dispatch.mock.calls.length).toBe(1)
      expect(dispatch.mock.calls[0]).toEqual([closePopover('pop1')])
    })
  })
})

/* eslint-enable
  import/no-extraneous-dependencies,
  react/jsx-filename-extension,
  react/no-children-prop,
  react/prop-types,
*/
