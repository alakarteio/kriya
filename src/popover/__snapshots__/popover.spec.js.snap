// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Popover container should pass onClose callback when closable 1`] = `
<div
  className="popover"
  onClick={[Function]}
  style={Object {}}
>
  <div>
    custom_component_child{"print":false}
  </div>
</div>
`;

exports[`common/Popover container should pass print=false when the popover is not listed in redux 1`] = `
<div
  className="popover"
  onClick={undefined}
  style={Object {}}
>
  <div>
    custom_component_child{"print":false}
  </div>
</div>
`;

exports[`common/Popover container should pass print=true when the popover IS listed in redux 1`] = `
<div
  className="popover print"
  onClick={undefined}
  style={Object {}}
>
  <div>
    custom_component_child{"print":true}
  </div>
</div>
`;

exports[`common/Popover graphical (JSX) should add children 1`] = `
<div
  className="popover"
  onClick={undefined}
  style={Object {}}
>
  <div>
    custom_2{"print":false}
  </div>
</div>
`;

exports[`common/Popover graphical (JSX) should add custom className 1`] = `
<div
  className="popover custom"
  onClick={undefined}
  style={Object {}}
>
  <div>
    custom_component_child{"print":false}
  </div>
</div>
`;

exports[`common/Popover graphical (JSX) should add custom style 1`] = `
<div
  className="popover"
  onClick={undefined}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <div>
    custom_component_child{"print":false}
  </div>
</div>
`;

exports[`common/Popover graphical (JSX) should be print when asked for and passed it to child 1`] = `
<div
  className="popover print"
  onClick={undefined}
  style={Object {}}
>
  <div>
    custom_component_child{"print":true}
  </div>
</div>
`;

exports[`common/Popover graphical (JSX) should have a default behaviour 1`] = `
<div
  className="popover"
  onClick={undefined}
  style={Object {}}
>
  <div>
    custom_component_child{"print":false}
  </div>
</div>
`;

exports[`common/Popover graphical (JSX) should pass onClose function to child 1`] = `
<div
  className="popover"
  onClick={[Function]}
  style={Object {}}
>
  <div>
    custom_component_child{"print":false}
  </div>
</div>
`;
